### PR TITLE
Refactor logs command

### PR DESCRIPTION
**Pull request by rosspeoples**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Refactors the log command to separate Options and Flags.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
I tried using `genericclioptions.ResourceFinder` instead of `*resource.Builder`, but the tests were failing due to the discovery client not being set on expander errors.

Also tried to remove `cmdutil.Factory` in favor of `RESTClientGetter`, but I needed the selected namespace and its `NewBuilder()` which works properly with `cmdtesting.NewTestFactory()`.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

Related issue: https://github.com/kubernetes/kubectl/issues/1046

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:103]: `ResourceBuilderFlags` is preferred rather than this, see https://github.com/kubernetes/kubernetes/blob/c75d254beb662ea73736d2f123d0382b7f91dfbd/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L88

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:131]: You don't need that value yet, and you're swallowing error, which is not acceptable.

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:146]: Why do you need this, it's a string flag so you can easily do `len(flags.Container) != 0`

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:134]: Changing this you're breaking the current contract, don't!

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold

**Comment by dims** [comment]
this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

**Comment by soltysh** [comment]
/close 
I'll be picking up in the near future.

**Pull request closed without merging**
